phpBB/posting.php

```diff
@@ -616,6 +616,7 @@
 if ($mode != 'edit')
 {
 	$post_data['enable_sig']		= ($config['allow_sig'] && $user->optionget('attachsig')) ? true: false;
+	$post_data['topic_status']		= ($mode == 'post' && (isset($_POST['lock_topic']) && $auth->acl_get('m_lock', $forum_id))) ? true : false;
```

Comment:
- Use `$request->is_set_post('lock_topic')` instead of `(isset($_POST['lock_topic'])`.
- Users can also lock their own topics. Use `$auth->acl_gets('m_lock', 'f_user_lock', $forum_id)))` to check for both permissions.
- topic_status is an integer, not a boolean. Use `ITEM_LOCKED` and `ITEM_UNLOCKED` to get the appropriate value.
- Move the this line after `$post_data['enable_urls']...`